### PR TITLE
docs: update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by johnbieren**
This commit updates the CONTRIBUTING.md for checkton, the internal resources sym links, and the data keys schema.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -316,3 +316,28 @@ PR paired with an e2e change (documented [here](https://github.com/redhat-appstu
...
+
+### Data Keys Schema
+
+This repository maintains a json schema for the data key used in various tasks in this repo. It is stored [here](schema/dataKeys.json).
+
+If your change adds or removes a key to the data file, the schema must be updated accordingly as part of your pull request.
+
+### Checkton check
+
+This repository uses [checkton](https://github.com/chmeliik/checkton) to run shellchecker on the embedded shell in the tekton resources.
```
Comment (mmalina) [inline CONTRIBUTING.md]: It's called `shellcheck` , maybe you even could link it: https://www.shellcheck.net

Also, probably better to say "on the embedded scripts", and Tekton with capital T.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -316,3 +316,28 @@ PR paired with an e2e change (documented [here](https://github.com/redhat-appstu
...
+
+This check shows itself as the `Linters / checkton (pull_request)` check on the pull request.
+
+If it fails and you click details, the tool does a pretty good job of highlighting the failures and telling you how to fix them.
+
+We strive to have all of our tekton resources abide by shellchecker, so this check is mandatory for pull requests submitted to this repo.
```
Comment (mmalina) [inline CONTRIBUTING.md]: Typo: shellcheck

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -316,3 +316,28 @@ PR paired with an e2e change (documented [here](https://github.com/redhat-appstu
...
+
+### Symlinks (internal dir changes only)
+
+This repository contains [one directory](internal/resources) for all tekton resources inside of the internal directory.
+
+This is so that we can point argo at it and have both the tasks and pipelines deployed to the cluster (but not the test files).
```
Comment (mmalina) [inline CONTRIBUTING.md]: argo -> ArgoCD

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -316,3 +316,28 @@ PR paired with an e2e change (documented [here](https://github.com/redhat-appstu
...
+
+If you submit a change that either adds or removes a pipeline from the [internal pipelines directory](internal/pipelines) or task from the [internal tasks directory](internal/tasks),
```
Comment (mmalina) [inline CONTRIBUTING.md]: or task -> or a task

File: CONTRIBUTING.md (hunk 1 above)
Comment (johnbieren) [reply CONTRIBUTING.md]: fixed and linked

File: CONTRIBUTING.md (hunk 2 above)
Comment (johnbieren) [reply CONTRIBUTING.md]: fixed

File: CONTRIBUTING.md (hunk 3 above)
Comment (johnbieren) [reply CONTRIBUTING.md]: fixed

File: CONTRIBUTING.md (hunk 4 above)
Comment (johnbieren) [reply CONTRIBUTING.md]: fixed

**Review by mmalina — APPROVED** [review]

**Review by happybhati — APPROVED** [review]

**Pull request merged**